### PR TITLE
DEPR: enforce not-automatically aligning in DataFrame comparisons

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -568,6 +568,7 @@ Removal of prior version deprecations/changes
 - Enforced deprecation of calling numpy "ufunc"s on :class:`DataFrame` with ``method="outer"``; this now raises ``NotImplementedError`` (:issue:`36955`)
 - Enforced deprecation disallowing passing ``numeric_only=True`` to :class:`Series` reductions (``rank``, ``any``, ``all``, ...) with non-numeric dtype (:issue:`47500`)
 - Changed behavior of :meth:`DataFrameGroupBy.apply` and :meth:`SeriesGroupBy.apply` so that ``group_keys`` is respected even if a transformer is detected (:issue:`34998`)
+- Comparisons between a :class:`DataFrame` and a :class:`Series` where the frame's columns do not match the series's index raise ``ValueError`` instead of automatically aligning, do ``left, right = left.align(right, axis=1, copy=False)`` before comparing (:issue:`46795`)
```
Comment (MarcoGorelli) [inline doc/source/whatsnew/v2.0.0.rst:571]: is this the right issue number?

**Review by MarcoGorelli — COMMENTED** [review]
Changes look good to me - just got a comment, and wanted to double-check about the issue number in the whatsnew

File: pandas/tests/arithmetic/test_datetime64.py [hunk 2]
```diff
@@ -307,43 +307,60 @@ def test_timestamp_compare_series(self, left, right):
 
     def test_dt64arr_timestamp_equality(self, box_with_array):
         # GH#11034
+        box = box_with_array
 
         ser = Series([Timestamp("2000-01-29 01:59:00"), Timestamp("2000-01-30"), NaT])
-        ser = tm.box_expected(ser, box_with_array)
+        ser = tm.box_expected(ser, box)
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_datetime64.py:313]: Yes, it’s a pattern we use elsewhere

File: doc/source/whatsnew/v2.0.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v2.0.0.rst:571]: Nope, will update. Thanks

**Review by MarcoGorelli — APPROVED** [review]

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Ah wait, the related ASV checks are failing
```
2022-11-29T00:54:09.9434614Z [  1.81%] ··· ...thSeriesAxis.time_frame_op_with_series_axis1        6/12 failed
2022-11-29T00:54:09.9435374Z [  1.81%] ··· ========== ==========
2022-11-29T00:54:09.9435738Z                 opname             
2022-11-29T00:54:09.9436168Z               ---------- ----------
2022-11-29T00:54:09.9436526Z                   eq       failed  
2022-11-29T00:54:09.9437238Z                   ne       failed  
2022-11-29T00:54:09.9437518Z                   lt       failed  
2022-11-29T00:54:09.9437736Z                   le       failed  
2022-11-29T00:54:09.9437960Z                   ge       failed  
2022-11-29T00:54:09.9438183Z                   gt       failed  
2022-11-29T00:54:09.9438611Z                  add      12.0±0ms 
2022-11-29T00:54:09.9438899Z                  sub      11.8±0ms 
2022-11-29T00:54:09.9439204Z                truediv    11.8±0ms 
2022-11-29T00:54:09.9439479Z                floordiv   42.0±0ms 
2022-11-29T00:54:09.9439765Z                  mul      11.6±0ms 
2022-11-29T00:54:09.9440049Z                  pow      31.0±0ms 
2022-11-29T00:54:09.9440264Z               ========== ==========
```

Good to have fixed the flakiness, I'd got so used to seeing that check fail I was going to just restart it, but this time it's not a false positive

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**